### DISCUSSION TITLE
How do I know when FSearch is *completely* done indexing?

### DISCUSSION THREAD

**Discussion by Zlarg** [cboxdoerfer/fsearch#494] [Q&A]
I'm indexing a large number of files.  Every time I think FSearch is finished it starts indexing more files.  Will it tell me when it's finally finished?   It would be nice to have a progress indicator.

**Comment by cboxdoerfer** [comment]
The status of the indexing is shown in the status bar along with a button to cancel it. Once indexing is finsihed the status bar will reflect that.

A progress bar hasn't been implemented and likely won't, because for it to be useful and somewhat accurate it would significantly slow down the indexing, as you need to walk the file system to count the number of files to be indexed before indexing can take place.

**Reply by Zlarg** [reply]
So I essentially can't know when its actually finished and I can trust that the results are complete before I start using them.

**Reply by cboxdoerfer** [reply]
> So I essentially can't know when its actually finished and I can trust that the results are complete before I start using them.

Of course you can know when its finished. When the status bar shows "Updating" it's updating the index (it also displays the current folder being indexed), when it's finished it's showing the number of indexed files instead.
